### PR TITLE
DIT-757 Refer journey changes

### PR DESCRIPTION
**Pull request by retroredge**

### CODE REVIEW

**Comment by gokyo** [comment]
👍

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case.scala.html [hunk 1]
```diff
@@ -19,23 +19,33 @@
...
-            <p> Before you reject this case, add a note explaining why you’re rejecting it. </p>
-            <p>The days elapsed count will stop at <span id="days-elapsed">@(c.daysElapsed)</span>.</p>
+            @contact_details(c)
+
+            <h3 class="heading-medium mt-0">Have you contacted the applicant and attached the email to this case?</h3>
 
             @helper.form(action = RejectCaseController.confirmRejectCase(c.reference), 'id -> "reject-case") {
                 @helper.CSRF.formField
 
+                @input_radiogroup(
```
Comment (ed0906) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case.scala.html]: Do you think there's any value in a common `input_yes_no` component?

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case.scala.html [hunk 2]
```diff
@@ -19,23 +19,33 @@
...
+                    field = form("state"),
+                    options = Seq(RadioOption("true", "Yes"), RadioOption("false", "No")),
+                    id = "reject-case-contacted_applicant"
```
Comment (ed0906) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case.scala.html]: `reject_case-contacted_applicant`

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case_error.scala.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    <div class="grid-row">
+        <div class="column-full">
+            <h2 class="heading-large mt-0">You must contact the applicant and attach a copy of the email to this case before you reject it</h2>
+
+            @contact_details(c)
+
+            <h3 class="heading-medium mt-45">Next Steps</h3>
+
+            <p>Go to <a href="@AttachmentsController.attachmentsDetails(c.reference)">case attachments</a>.</p>
```
Comment (ed0906) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case_error.scala.html]: Does this link need an ID?

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case_error.scala.html [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+@import uk.gov.hmrc.tariffclassificationfrontend.controllers.routes.AttachmentsController
+@(c: Case)(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
+
+@main(bodyClasses = None) {
+
+    @case_heading(c)
+
+    <div class="grid-row">
+        <div class="column-full">
+            <h2 class="heading-large mt-0">You must contact the applicant and attach a copy of the email to this case before you reject it</h2>
```
Comment (ed0906) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case_error.scala.html]: Does this header need an ID?

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/refer_case_error.scala.html [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+@import uk.gov.hmrc.tariffclassificationfrontend.controllers.routes.AttachmentsController
+@(c: Case)(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
+
+@main(bodyClasses = None) {
+
+    @case_heading(c)
+
+    <div class="grid-row">
+        <div class="column-full">
+            <h2 class="heading-large mt-0">You must contact the applicant and attach a copy of the email to this case before you refer it</h2>
```
Comment (ed0906) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/refer_case_error.scala.html]: Do we need some more IDs for the acceptance tests here?

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/refer_case_error.scala.html [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+            @contact_details(c)
+
+            <h3 class="heading-medium mt-45">Next Steps</h3>
+
+            <p>Go to <a href="@AttachmentsController.attachmentsDetails(c.reference)">case attachments</a>.</p>
```
Comment (ed0906) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/refer_case_error.scala.html]: Do we need some more IDs for the acceptance tests here?

File: app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ReferCaseController.scala [hunk 7]
```diff
@@ -35,18 +37,32 @@ class ReferCaseController @Inject()(authenticatedAction: AuthenticatedAction,
...
+
   override protected val config: AppConfig = appConfig
   override protected val caseService: CasesService = casesService
 
   override protected def redirect: String => Call = routes.CaseController.applicationDetails
   override protected def isValidCase(c: Case)(implicit request: AuthenticatedRequest[_]): Boolean = c.status == OPEN
 
   def referCase(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
-    getCaseAndRenderView(reference, c => successful(views.html.refer_case(c)))
+    getCaseAndRenderView(reference, c => successful(views.html.refer_case(c, form)))
```
Comment (gokyo) [inline app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ReferCaseController.scala]: this line could be extracted as 
```
 def showCase(reference: String, c: Case, f: Form[Boolean]) = {
     getCaseAndRenderView(reference, c => successful(views.html.refer_case(c, f)))
}
```
Then,
At line 49 you can call `showCase(reference, c, form)`
At line 56 you can call `showCase(reference, c, errors)`

Comment (gokyo) [reply app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ReferCaseController.scala]: There is something similar in `CancelRulingController`

File: app/uk/gov/hmrc/tariffclassificationfrontend/controllers/RejectCaseController.scala [hunk 8]
```diff
@@ -35,18 +37,31 @@ class RejectCaseController @Inject()(authenticatedAction: AuthenticatedAction,
...
+
   override protected val config: AppConfig = appConfig
   override protected val caseService: CasesService = casesService
 
   override protected def redirect: String => Call = routes.CaseController.applicationDetails
   override protected def isValidCase(c: Case)(implicit request: AuthenticatedRequest[_]): Boolean = c.status == OPEN
 
   def rejectCase(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
-    getCaseAndRenderView(reference, c => successful(views.html.reject_case(c)))
+    getCaseAndRenderView(reference, c => successful(views.html.reject_case(c, form)))
```
Comment (gokyo) [inline app/uk/gov/hmrc/tariffclassificationfrontend/controllers/RejectCaseController.scala]: line 49 and line 55 can re-use a common method

File: app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuspendCaseController.scala [hunk 9]
```diff
@@ -35,18 +37,31 @@ class SuspendCaseController @Inject()(authenticatedAction: AuthenticatedAction,
...
   def suspendCase(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
-    getCaseAndRenderView(reference, c => successful(views.html.suspend_case(c)))
+    getCaseAndRenderView(reference, c => successful(views.html.suspend_case(c, form)))
   }
 
   def confirmSuspendCase(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
-    getCaseAndRenderView(reference, casesService.suspendCase(_, request.operator).map(views.html.confirm_with_contact_details(_, "suspended")))
+    form.bindFromRequest().fold(
+      errors => {
+        getCaseAndRenderView(
```
Comment (gokyo) [inline app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuspendCaseController.scala]: line 49 and line 55 can reuse a common method

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case.scala.html (hunk 1 above)
Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case.scala.html]: Yes. If I have time I will try and extract one

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case.scala.html (hunk 2 above)
Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case.scala.html]: I'd prefer reject-case_contacted-applicant :)

Maybe we need a team huddle to agree a standard

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case_error.scala.html (hunk 3 above)
Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case_error.scala.html]: Done

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case_error.scala.html (hunk 4 above)
Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case_error.scala.html]: Done

Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case_error.scala.html]: Ash says `headings are fine as is`

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/refer_case_error.scala.html (hunk 5 above)
Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/refer_case_error.scala.html]: Ash says `headings are fine as is`

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/refer_case_error.scala.html (hunk 6 above)
Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/refer_case_error.scala.html]: Done

File: app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ReferCaseController.scala (hunk 7 above)
Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/controllers/ReferCaseController.scala]: As discussed I'm going to leave this as it's already readable

File: app/uk/gov/hmrc/tariffclassificationfrontend/controllers/RejectCaseController.scala (hunk 8 above)
Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/controllers/RejectCaseController.scala]: As discussed I'm going to leave this as it's already readable

File: app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuspendCaseController.scala (hunk 9 above)
Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuspendCaseController.scala]: As discussed I'm going to leave this as it's already readable

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case.scala.html (hunk 1 above)
Comment (retroredge) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/reject_case.scala.html]: Done

**Pull request merged**